### DISCUSSION TITLE
How to run on raspberry pi. Answered && Closed

### DISCUSSION THREAD

**Discussion by ccdanieldb** [mrlt8/docker-wyze-bridge#95] [Q&A]
I have been trying to get this running on a raspi4 with no luck.  My setup is listed below

`~/docker-wyze-bridge $ uname -a
Linux 5.10.52-v7l+ #1441 SMP Tue Aug 3 18:11:56 BST 2021 armv7l GNU/Linux`

`~/docker-wyze-bridge $ docker --version
Docker version 20.10.8, build 3967b7d`

I have listed the steps I have taken below.  

I edited my docker-compose.yml as below.

        version: '2.4' 
        services:
                wyze-bridge:
                        container_name: wyze-bridge
                        restart: always
                        image: mrlt8/wyze-bridge:latest # Use a prebuilt image
                        # build:  # Uncomment to build from source
                        #     context: ./app # Uncomment to build from source
                        dockerfile: Dockerfile.arm # Uncomment to build for arm

                        ports:
                                    - 1935:1935
                                    - 8554:8554
                                    - 8888:8888

                        environment:
                                    - WYZE_EMAIL=${notmy@real.email} # Replace with wyze email
                                    - WYZE_PASSWORD=${notmypassword} # Replace with wyze password

Then I ran the below command in the /docker-wyze-bridge directory via ssh.  

`docker run docker-compose up --build`

and this is what it returned.  

        ~/docker-wyze-bridge $ docker run docker-compose up --build

        Unable to find image docker-compose:latest locally

        docker: Error response from daemon: pull access denied for docker-compose, repository does not exist or may require                         'docker login': denied: requested access to the resource is denied.

        See docker run --help.`

I am sure I am missing something, but do not know where to start looking.  This is my first experience with docker as well.

**Comment by mrlt8** [comment]
The prebuilt image is multi-arch, so you don't need `dockerfile: Dockerfile.arm` (you can delete or comment that line out).
You only need that line if you're building from source.

**Reply by ccdanieldb** [reply]
How do I build from source?  In the "ARM/Raspberry Pi Support" it has the below language for running on arm. 

The default configuration will use the x64 tutk library, however, you can edit your docker-compose.yml to use the 32-bit arm library by setting dockerfile as Dockerfile.arm:

**Reply by mrlt8** [reply]
The image should be listed as an alternate option for arm, but if you'd like to build from source, you need to delete or comment out the image line and uncomment the _three_ line for building:
```yaml
    build:
        context: ./app
        dockerfile: Dockerfile.arm
```

**Reply by ccdanieldb** [reply]
Thank you mrlt8

After installing docker-compose because for some reasion (curl -sSL https://get.docker.com | sh) dose not install it.  This worked but I am still having the below issue now with build.  I will close this Q&A because the problem has been answered allowing me to proceed with the build command.

**Comment by ccdanieldb** [comment]
Ok just realized that docker compose is not installed along with docker.

**Comment by ccdanieldb** [comment]
Ok I installed docker-compose and updated my docker-compose file like below and can now get a build going.  

   `version: '2.4'

    services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: always
        #image: mrlt8/wyze-bridge:latest # Use a prebuilt image
        build:  # Uncomment to build from source
            context: ./app # Uncomment to build from source
            dockerfile: Dockerfile.arm # Uncomment to build for arm
        ports:
            - 1935:1935
            - 8554:8554
            - 8888:8888
        environment:
            - WYZE_EMAIL= # Replace with wyze email
            - WYZE_PASSWORD= # Replace with wyze password`

But am still having issues.  

Durning the build process I get several errors resulting the the build proses failing.  

	~/docker-wyze-bridge $ sudo docker-compose up --build
	Building wyze-bridge
	Sending build context to Docker daemon  31.74kB
	Step 1/19 : ARG ARM=1
	Step 2/19 : ARG ARCH=${ARM:+arm32v7}
	Step 3/19 : FROM ${ARCH:-amd64}/python:3.9-slim as base
	 ---> 6b23301db870
	Step 4/19 : FROM base as builder
	 ---> 6b23301db870
	Step 5/19 : ENV PYTHONUNBUFFERED=1
	 ---> Using cache
	 ---> 450732dbd869
	Step 6/19 : ARG ARM
	 ---> Using cache
	 ---> 226d6e834e78
	Step 7/19 : ARG TUTK_ARCH=${ARM:+Arm11_BCM2835_4.8.3}
	 ---> Using cache
	 ---> c9e696bd5603
	Step 8/19 : ARG RTSP_ARCH=${ARM:+armv7}
	 ---> Using cache
	 ---> 78ead2b8412d
	Step 9/19 : ARG FFMPEG_ARCH=${ARM:+armhf}
	 ---> Using cache
	 ---> e61428d77d0c
	Step 10/19 : RUN apt-get update     && apt-get install -y tar xz-utils unzip curl jq 			g++     && apt-get clean     && rm -rf /var/lib/apt/lists/*
 	---> Running in aa35c7a3d31a
	Get:1 http://deb.debian.org/debian bullseye InRelease [113 kB]
	Get:2 http://security.debian.org/debian-security bullseye-security InRelease [44.1 kB]
	Err:2 http://security.debian.org/debian-security bullseye-security InRelease
	  At least one invalid signature was encountered.
	Err:1 http://deb.debian.org/debian bullseye InRelease
	  At least one invalid signature was encountered.
	Get:3 http://deb.debian.org/debian bullseye-updates InRelease [36.8 kB]
	Err:3 http://deb.debian.org/debian bullseye-updates InRelease
	  At least one invalid signature was encountered.
	Reading package lists...
	W: GPG error: http://security.debian.org/debian-security bullseye-security InRelease: 	At least one invalid signature was encountered.
	E: The repository 'http://security.debian.org/debian-security bullseye-security InRelease' is not signed.
	W: GPG error: http://deb.debian.org/debian bullseye InRelease: At least one invalid signature was encountered.
	E: The repository 'http://deb.debian.org/debian bullseye InRelease' is not signed.
	W: GPG error: http://deb.debian.org/debian bullseye-updates InRelease: At least one invalid signature was encountered.
	E: The repository 'http://deb.debian.org/debian bullseye-updates InRelease' is not signed.
	The command '/bin/sh -c apt-get update     && apt-get install -y tar xz-utils unzip curl jq g++     && apt-get clean     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100
	ERROR: Service 'wyze-bridge' failed to build : Build failed
	pi@ar8-pi4-8:~/docker-wyze-bridge $

**Reply by mrlt8** [reply]
`docker system prune`?

**Reply by ccdanieldb** [reply]
I open a seperat Q&A about this because it is a separate issued.  please comment on "docker-compose up --build on raspberry pi 4 is failing". Also do you know how to close an issue.
